src/client/options/mod.rs

```diff
@@ -718,6 +825,37 @@ impl TlsOptions {
 
         Ok(config)
     }
+
+    #[cfg(test)]
+    pub(crate) fn serialize_for_client_options<S>(
+        tls_options: &TlsOptions,
+        serializer: S,
+    ) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        #[derive(Serialize)]
+        struct TlsOptionsHelper<'a> {
+            tls: Option<bool>,
+            tlscafile: Option<&'a str>,
+            tlscertificatekeyfile: Option<&'a str>,
+            tlsallowinvalidcertificates: Option<bool>,
+        }
+
+        let state = TlsOptionsHelper {
+            tls: tls_options.ca_file_path.as_ref().map(|_| true),
```

Comment:
I think this does not need to be conditional, since a `TlsOptions` is only ever serialized from the `Tls::Enabled` enum variant.

Reply:
Let me know what you think of my new implementation.